libminifi/include/utils/StringUtils.h

```diff
@@ -46,43 +47,53 @@ namespace utils {
 /**
  * Stateless String utility class.
  *
- * Design: Static class, with no member variables
+ * Design: class, with no member variables
  *
  * Purpose: Houses many useful string utilities.
  */
-class StringUtils {
- public:
+
+namespace {
+  template<class Char>
+  struct string_traits;
+  template<>
+  struct string_traits<char>{
+    template<class T>
+    static std::string convert_to_string(T&& t){
+      return std::to_string(std::forward<T>(t));
+    }
+  };
+
+  template<>
+  struct string_traits<wchar_t>{
+    template<class T>
+    static std::wstring convert_to_string(T&& t){
+      return std::to_wstring(std::forward<T>(t));
+    }
+  };
+}
+
+namespace StringUtils {
   /**
    * Converts a string to a boolean
    * Better handles mixed case.
    * @param input input string
    * @param output output string.
    */
-  static bool StringToBool(std::string input, bool &output) {
```

Comment:
As I suspected this caused issues. We're giving weak guidance to the linker. So this caused issues when running some extensions that are loaded dynamically ( where the local linking guidance causes no issues ) -- so while this isn't a bad, that must be resolved first.